src/ophys_etl/workflows/pipeline_modules/trace_extraction.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from types import ModuleType
+from typing import List, Dict
+
+import json
+
+from ophys_etl.modules import segment_postprocess
+from ophys_etl.workflows.workflow_steps import WorkflowStepEnum
+from sqlmodel import Session
+
+from ophys_etl.workflows.ophys_experiment import OphysExperiment
+
+from ophys_etl.workflows.output_file import OutputFile
+from ophys_etl.workflows.pipeline_module import PipelineModule
+from ophys_etl.workflows.well_known_file_types import WellKnownFileTypeEnum
+
+class TraceExtractionModule(PipelineModule):
+    """Trace extraction module"""
+    def __init__(
+        self,
+        ophys_experiment: OphysExperiment,
+        prevent_file_overwrites: bool = True,
+        **kwargs
+    ):
+        super().__init__(
+            ophys_experiment=ophys_experiment,
+            prevent_file_overwrites=prevent_file_overwrites,
+            **kwargs
+        )
+
+        motion_corrected_ophys_movie_file: OutputFile = \
+            kwargs['motion_corrected_ophys_movie_file']
+        self._motion_corrected_ophys_movie_file = str(motion_corrected_ophys_movie_file.path)
+
+    @property
+    def queue_name(self) -> WorkflowStepEnum:
+        return WorkflowStepEnum.TRACE_EXTRACTION
+
+    @property
+    def inputs(self, session: Session) -> Dict:
```

Comment:
1) I don't think it's a good pattern to pass arguments to `@property`. They are supposed to act like attributes, so be able to do `.inputs` without passing arguments. 
2) `session` will not be passed

Better to encapsulate this logic in `OphysExperiment` class so you can just do `self.ophys_experiment.motion_border`. So in `motion_border` property, do 
```
with Session(engine) as session:
    ...
```

Same goes for `get_ophys_experiment_roi_metadata`. Better to do `self.ophys_experiment.rois` and encapsulate logic there to pull a session, and make it a property rather than method.

Reply:
I've been finding it challenging to write tests for this. The usual method mock engine as `cls._engine`. This can be passed in when initializing Session. 

With Session initialized with engine, I tried [patching engine with self._engine.](https://github.com/AllenInstitute/ophys_etl_pipelines/blob/1f3967df807b66678ac728b0b0d1e84a8a935cb2/tests/workflows/test_ophys_experiment.py#L104)
However, this has not worked in that the method I'm testing was not able to access the [mock data that I set up.](https://github.com/AllenInstitute/ophys_etl_pipelines/blob/1f3967df807b66678ac728b0b0d1e84a8a935cb2/tests/workflows/test_ophys_experiment.py#L40)

I also tried patching the Session as well.

Reply:
try patching `ophys_etl.workflows.ophys_experiment.engine`